### PR TITLE
simple: Define SIMPLE_SKIPPED_INIT_CAPS better

### PR DESCRIPTION
**Pull request by Furao**
Make sure this macro is defined based on all possible configurations.

Required based on this change: https://github.com/seL4/seL4/pull/1088

### CODE REVIEW

File: libsel4simple/arch_include/arm/simple/arch/simple.h [hunk 1]
```diff
@@ -16,7 +16,13 @@
 
 /* Simple does not address initial null caps, including seL4_CapNull.
  * seL4_CapIOSpace, seL4_CapIOPortControl are null on architectures other than x86 */
-#define SIMPLE_SKIPPED_INIT_CAPS 3
+#ifdef CONFIG_KERNEL_MCS
+#define SKIP_THREADSC 0
```
Comment (axel-h) [inline libsel4simple/arch_include/arm/simple/arch/simple.h:20]: Maybe better prefix all the new defined with `SIMPLE_` to have a kind of namespace.

Comment (Furao) [reply libsel4simple/arch_include/arm/simple/arch/simple.h:20]: Good idea. I will do that.

Comment (Furao) [reply libsel4simple/arch_include/arm/simple/arch/simple.h]: Done

**Comment by kent-mcleod** [comment]
Thanks for putting this PR up @Furao ,

I think `simple_default_nth_cap` needs to also be updated to skip over the null caps correctly. It looks like it currently only handles the special x86 caps at the moment.

I think this also needs to account for seL4_CapSMMUSIDControl and seL4_CapSMMUCBControl caps which would be null if SMMUv2 isn't enabled.

**Comment by Furao** [comment]
> Thanks for putting this PR up @Furao ,

No Problem

> I think `simple_default_nth_cap` needs to also be updated to skip over the null caps correctly. It looks like it currently only handles the special x86 caps at the moment.
> 
> I think this also needs to account for seL4_CapSMMUSIDControl and seL4_CapSMMUCBControl caps which would be null if SMMUv2 isn't enabled.

Ok, I updated that function. I looked at that a bit ago when Axel first brought this issue up on the SMC patches, but I was confused by this function. That explains why.

**Review by kent-mcleod — APPROVED** [review]

**Pull request merged**